package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,458 @@
+            'angstrom': 'Angstrom',
+            'A': 'Angstrom',
+            'nm': 'nm',
+            'pm': 'pm',
+            'fm': 'fm',
+            },
+        'velocity': {
+            'Angstrom ps-1': 'Angstrom/ps',
+            'A ps-1': 'Angstrom/ps',
+            'Angstrom fs-1': 'Angstrom/fs',
+            'A fs-1': 'Angstrom/fs',
+            'Angstrom AKMA-1': 'Angstrom/AKMA',
+            'A AKMA-1': 'Angstrom/AKMA',
+            'nm ps-1': 'nm/ps',
+            'nm ns-1': 'nm/ns',
+            'pm ps-1': 'pm/ps',
+            'm s-1': 'm/s'
+            },
+        'force':  {
+            'kJ mol-1 Angstrom-1': 'kJ/(mol*Angstrom)',
+            'kJ mol-1 nm-1': 'kJ/(mol*nm)',
+            'Newton': 'Newton',
+            'N': 'N',
+            'J m-1': 'J/m',
+            'kcal mol-1 Angstrom-1': 'kcal/(mol*Angstrom)',
+            'kcal mol-1 A-1': 'kcal/(mol*Angstrom)'
+            }
+    }
+    _Timestep = Timestep
+    _data_keywords = ('time', 'step', 'lambda', 'dt')
+
+    def __init__(self, filename, convert_units=True, **kwargs):
+        """
+        Parameters
+        ----------
+        filename : str or :class:`h5py.File`
+            trajectory filename or open h5py file
+        convert_units : bool (optional)
+            convert units to MDAnalysis units
+        **kwargs : dict
+            General reader arguments.
+
+        Raises
+        ------
+        RuntimeError
+            when `H5PY`_ is not installed
+        RuntimeError
+            when a unit is not recognized by MDAnalysis
+        ValueError
+            when ``n_atoms`` changes values between timesteps
+
+        """
+        if not HAS_H5PY:
+            raise RuntimeError("Please install h5py")
+        super(H5MDReader, self).__init__(filename, **kwargs)
+        self.filename = filename
+        self.open_trajectory()
+        self.has_positions = 'position' in self._particle_group
+        if not self.has_positions:
+            raise ValueError("'position' group must be in 'particles' group")
+        self.n_atoms = self._particle_group['position/value'].shape[1]
+        self.has_velocities = 'velocity' in self._particle_group
+        self.has_forces = 'force' in self._particle_group
+        self.ts = self._Timestep(self.n_atoms,
+                                 velocities=self.has_velocities,
+                                 forces=self.has_forces,
+                                 **self._ts_kwargs)
+        self.units = {'time': None,
+                      'length': None,
+                      'velocity': None,
+                      'force': None}
+        self._translate_h5md_units()  # fills units dictionary
+        self._read_next_timestep()
+
+    @staticmethod
+    def _format_hint(thing):
+        """Can this Reader read *thing*"""
+        # nb, filename strings can still get passed through if
+        # format='H5MD' is used
+        return HAS_H5PY and isinstance(thing, h5py.File)
+
+    def open_trajectory(self):
```

Comment:
This might need to keep kwargs from the reader such as `comm=MPI.COMM_WORLD` and `driver="mpio"` (http://docs.h5py.org/en/stable/mpi.html).

Importantly, this needs to work for reopening a trajectory, too.